compiler/src/middle_end/optimize_constants.re

```diff
@@ -3,7 +3,7 @@ open Grain_typed;
 open Grain_utils;
 open Types;
 
-let known_constants = ref(Ident.empty: Ident.tbl(constant));
+let known_constants = ref(Ident.empty: Ident.tbl(imm_expression_desc));
```

Comment:
We weren't previously optimizing a program like
```
let y = x
y
```
into
```
x
```
and it's needed for the compiler to do other optimizations on simple rebinds, like `let (+) = WasmI32.add`. (This worked previously for import aliases, i.e. `import { add as (+) } ...` because they're handled differently than regular bindings.)

Reply:
_Note:_ This optimization is only performed on immutable bindings. We don't yet have an optimization that rewrites `let mut`s into `let`s when no mutation is performed.